Cabal/src/Distribution/Simple/BuildToolDepends.hs

```diff
@@ -13,6 +13,26 @@ import qualified Data.Map as Map
 import           Distribution.Package
 import           Distribution.PackageDescription
 
+-- | Same as 'desugarBuildTool', but requires atomic informations (package
+-- name, executable names) instead of a whole 'PackageDescription'.
+desugarBuildToolSimple :: PackageName
+                          -> [UnqualComponentName]
+                          -> LegacyExeDependency
+                          -> Maybe ExeDependency
+desugarBuildToolSimple pname exeNames led =
+  if foundLocal
+  then Just $ ExeDependency pname toolName reqVer
+  else Map.lookup name whiteMap
+  where
+    LegacyExeDependency name reqVer = led
+    toolName = mkUnqualComponentName name
+    foundLocal = toolName `elem` exeNames
+    whitelist = [ "hscolour", "haddock", "happy", "alex", "hsc2hs", "c2hs"
+                , "cpphs", "greencard", "hspec-discover"
+                ]
+    whiteMap  = Map.fromList $ flip map whitelist $ \n ->
+      (n, ExeDependency (mkPackageName n) (mkUnqualComponentName n) reqVer)
+
 -- | Desugar a "build-tools" entry into proper a executable dependency if
```

Comment:
Maybe fix that `a` in the comment while you're on this. I guess, they mean "into a proper executable..."